### PR TITLE
Modify DnResolver changes for API compatibility.

### PR DESCRIPTION
**Pull request by dfish3r**
Added DnResolverEx interface and User class.
Updated applicable DnResolver implementations to implement both DnResolver and DnResolverEx.

### CODE REVIEW

File: core/src/main/java/org/ldaptive/auth/AuthenticationRequest.java [hunk 1]
```diff
@@ -71,9 +110,20 @@ public Object getUser()
...
+  public void setUser(final String id)
+  {
+    user = new User(id);
+  }
+
+
+  /**
+   * Sets the user.
+   *
+   * @param  u  of the user
```
Comment (serac) [inline core/src/main/java/org/ldaptive/auth/AuthenticationRequest.java]: The parameter javadoc makes no sense. Should be "the user to authenticate".

**Comment by serac** [comment]
I noted that `FormatDnResolver` doesn't implement the new `DnResolverEx` interface. Was that intentional?

**Comment by dfish3r** [comment]
I didn't have another impl that needed the new functionality. But I'll add it for consistency.

File: core/src/main/java/org/ldaptive/auth/AuthenticationRequest.java (hunk 1 above)
Comment (dfish3r) [inline core/src/main/java/org/ldaptive/auth/AuthenticationRequest.java]: I'll fix.

**Comment by dfish3r** [comment]
Pushed some changes, merge if you're satisfied.

**Pull request merged**